core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -193,6 +194,13 @@ public RemoteClusterService getRemoteClusterService() {
         return transportService.getRemoteClusterService();
     }
 
+    /**
+     * Return a map of nodeId to pending number of requests for the given action name prefix
+     */
+    public Map<String, Long> getPendingRequests(final String actionNamePrefix) {
```

Comment:
this seems to be a very expensive operation I wonder if we should special case this  here rather than adding a generic way of doing this.